liquibase-core/src/main/java/liquibase/change/core/AddAutoIncrementChange.java

```diff
@@ -60,7 +65,7 @@ public void setSchemaName(String schemaName) {
         this.schemaName = schemaName;
     }
 
-    @DatabaseChangeProperty(mustEqualExisting = "column.relation")
```

Comment:
@nvoxland (cc: @adrian-velonis1 - @filipelautert  ) Another question I have is about "mustEqualExisting" property. It says it adds the specified column into changeParameterMetaData, what is it used for? Is it also reflected in the change documentation page as well?

Reply:
This one has always been mysterious to me, but my understanding is that `mustEqualExisting` tells the application how the value of the attribute _ought_ to be applied when applying the change to an existing high-level object in the database—there's potentially room for error. So in `AddAutoIncrementChange`, the value returned by `getTableName()` corresponds to `column.relation` since you're adding the increment on a particular column in a particular table. I believe this is supposed to be Liquibase's way of providing a generic "path" for modifying the object in whichever database you're using.

Nathan might be able to explain in more detail and/or correct me if I've misunderstood.

As far as I know, `mustEqualExisting` is only needed on attributes which:

1. Interact with objects, so we also don't use it for a boolean/whatever that defines some kind of general behavior
2. Interact with objects that, a high level, realistically exist on most/all databases (like tables)
3. Interact with _existing_ objects, so we do use it for `getTableName()` on `AddColumnChange`, but we don't use it for `getTableName()` on `CreateTableChange`

We probably wouldn't need to put this metadata on docs.liquibase.com, but it would be appropriate on Javadocs or contribute.liquibase.com. I don't have a strong opinion either way.